### PR TITLE
Unpin mxnet

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Unpin `mxnet`.

## How is this patch tested?

Existing + fixed unit tests.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Found an issue.

```
% python -c "import mxnet; print(mxnet.__version__)"
1.7.0
                                                                                                                                  
% pip show mxnet
Name: mxnet
Version: 1.7.0.post1
Summary: MXNet is an ultra-scalable deep learning framework. This version uses openblas and MKLDNN.
Home-page: https://github.com/apache/incubator-mxnet
Author: None
Author-email: None
License: Apache 2.0
Location: /Users/harutakakawamura/.pyenv/versions/miniconda3-4.7.12/envs/mlflow-dev-env/lib/python3.6/site-packages
Requires: requests, contextvars, graphviz, numpy
Required-by: 
```

For mxnet 1.7.0, the version on PyPI doesn't match (`1.7.0.post1`) `mxnet.__version__` ( 1.7.0).

File: tests/gluon/test_gluon_model_export.py [hunk 1]
```diff
@@ -213,6 +217,7 @@ def test_gluon_model_serving_and_scoring_as_pyfunc(gluon_model, model_data):
         model_uri=model_uri,
         data=pd.DataFrame(test_data.asnumpy()),
         content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON_SPLIT_ORIENTED,
+        extra_args=["--no-conda"],
```
Comment (harupy) [inline tests/gluon/test_gluon_model_export.py:220]: A workaround for https://github.com/mlflow/mlflow/pull/3786#issuecomment-739682036

Comment (harupy) [reply tests/gluon/test_gluon_model_export.py:222]: This change also allows us to run this test against the dev verson of `mxnet`.

**Review by dbczumar — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
In mxnet 1.6.0, this assertion fails:

```python
assert "training softmaxcrossentropyloss" in data.metrics
```

In mxnet 1.6.0, loss is not recorded in `estimator.metrics`.

```
# mxnet 1.5.0
>>> print(estimator.train_metrics)
train accuracy: 0.0960, train softmaxcrossentropyloss: 2.3017, validation accuracy: nan, validation softmaxcrossentropyloss: nan

# mxnet >= 1.6.0
>>> print(estimator.train_metrics)
[<mxnet.metric.Accuracy object at 0x7fa5302ef940>]
```

**Comment by harupy** [comment]
In mxnet > 1.6.0, `loss` is monitored only when `self._train_metrics` is falsey (meaning `train_metrics` in `Estimator.__init__`)

```python
    def _add_default_training_metrics(self):
        if not self._train_metrics:
            suggested_metric = _suggest_metric_for_loss(self.loss)
            if suggested_metric:
                self._train_metrics = [suggested_metric]
            loss_name = type(self.loss).__name__
            self._train_metrics.append(metric_loss(loss_name))  🔥

        for metric in self._train_metrics:
            # add training prefix to the metric name
            # it is useful for event handlers to distinguish them from validation metrics
            metric.name = 'training ' + metric.name
```

https://github.com/apache/incubator-mxnet/blob/b1b50210f3a9befe421063d3878d43d03c15cf1f/python/mxnet/gluon/contrib/estimator/estimator.py#L240-L260

File: tests/gluon_autolog/test_gluon_autolog.py [hunk 2]
```diff
@@ -62,10 +80,20 @@ def gluon_random_data_run(log_models=True):
...
+    # In mxnet >= 1.6.0, `Estimator` doesn't monitor `loss` if it's instantiated with
+    # `train_metrics`.
+    #
+    # ```
+    # Estimator(loss=SomeLoss()) -> monitors `loss`
+    # Estimator(loss=SomeLoss(), train_metrics=SomeMetrics()) -> doesn't monitor `loss`
+    # ```
+    if LooseVersion(mx.__version__) < LooseVersion("1.6.0"):
+        assert "{} softmaxcrossentropyloss".format(train_prefix) in data.metrics
+        assert "validation softmaxcrossentropyloss" in data.metrics
```
Comment (harupy) [inline tests/gluon_autolog/test_gluon_autolog.py:96]: Should we fix `gluon.autolog` to log `loss` regardless of the version of `mxnet`?

Comment (harupy) [reply tests/gluon_autolog/test_gluon_autolog.py]: Not sure if this is an intended behavior for `mxnet` though.

Comment (harupy) [reply tests/gluon_autolog/test_gluon_autolog.py]: I'll post a question in the mxnet repo.

Comment (dbczumar) [reply tests/gluon_autolog/test_gluon_autolog.py]: If `loss` is only produced when users don't specify any manual metrics, I think it makes sense to continue adopting that convention. I.e. we shouldn't force the autologging integration to emit `loss`. Instead, we should change the test.

**Pull request merged**